### ISSUE TITLE
`code` displays strange behavior with line-wrapped text

### ISSUE THREAD

**Issue by platformer** [platformer/typst-algorithms#10]
Line wrapping in `code` causes indent guides to look incorrect and causes lines to be cut off.

````typst
#code(
  indent-guides: 1pt + black
)[
  ```py
  def floyd_warshall(G):
    # let G be an adjacency matrix
    dist = G
    
    for k in range(len(G)):
      for i in range(len(G)):
        for j in range(len(G)):
          if dist[i][j] > dist[i][k] + dist[k][j]:
            dist[i][j] = dist[i][k] + dist[k][j]
            blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
    
    return dist
  ```
]
````

![image](https://github.com/platformer/typst-algorithms/assets/40146328/21399c3b-f6c0-4ecd-8bfa-e6dd3ab7cf8d)

**Comment by platformer** [comment]
This issue occurs because there is no simple way to tell if content is line wrapped (calls to `measure` assume infinite available width). This issue could be resolved if Typst's `layout` function is updated to return the actual size of a table cell.

This issue could also be resolved if tables supported arbitrary cell borders, because then I wouldn't have to deal with `measure` to draw indent guides of the correct length. There is an experimental branch that uses tablex to achieve this: https://github.com/platformer/typst-algorithms/tree/tablex. However, there are various bugs that prevent this branch from being merged. I'll revisit this issue once tablex hits v0.1.0.

**Comment by platformer** [comment]
I was able to adapt the solution in the tablex branch to the existing code used on main. Fixed as of 5bc82eb4299b8360c29e602ad229f5ebbaf4bf78.
